src/irmin-pack/pack.ml

```diff
@@ -159,6 +160,8 @@ struct
           let t = unsafe_v ?fresh ?readonly ?lru_size ~index root in
           Lwt.return t)
 
+    let invalidate ~readonly root = Hashtbl.remove roots (root, readonly)
```

Comment:
unrelated to that PR but it's really confusing to have so many level of caching.